README.md

```diff
@@ -1,45 +1,130 @@
+GUI for `zcat | gpg --verify | dd`.
+
+[rufus]: https://github.com/endlessm/rufus
+[gis]: https://gitlab.gnome.org/gnome/gnome-initial-setup
+[endlessm-gis]: https://github.com/endlessm/gnome-initial-setup
+
+
+Installing from Live Image
+--------------------------
+
+When booted into an Endless OS live system, eos-installer can be used to write
+a copy of the running (but unmodified) OS image to permanent storage.  To allow
+this, eos-installer is shipped as part of the normal Endless OS ostree, but is
+hidden from the end user unless the OS is booted in live mode.
+
+During the `gnome-initial-setup` flow, the user is offered the choice to try a
+live session, or run the reformatter immediately. If they choose the former,
+this application can be launched as a normal app from the live session.
+
+There are two disk layouts for live systems:
+
+1. A large exFAT partition with label `eoslive`, which contains an uncompressed
+   Endless OS image file and associated signature, next to the normal EFI and
+   BIOS boot partitions. We sometimes refer to this layout as a *combined live
+   + installer USB*. It's created using [`eos-write-live-image`][ewli]. 
+2. An ISO image, produced by [eos-image-builder][eib] as part of the OS build
+   process. Ignoring the bootloader stuff, this is an ISO9660 filesystem which
+   contains a SquashFS disk image containing an Endless OS disk image, and
+   signatures for both the uncompressed disk image and the SquashFS image
+   alongside it.
+
+These are both treated very similarly by `eos-installer`. Once it has located
+the partition containing the GPG signature for the uncompressed image, it reads
+the disk image itself from the read-only device-mapper device that the system
+is booted from. This avoids `eos-installer` needing to know how to extract a
+file from a SquashFS image. It's also convenient during development because you
+can basically ignore the SquashFS case.
+
+Because the `eos-installer` application is run from *within the disk image
+being installed*, this configuration is quite hard to test during development:
+replacing the `eos-installer` binary modifies the disk image and so invalidates
+its GPG signature.
+
+[ewli]: https://github.com/endlessm/eos-meta/blob/master/eos-tech-support/eos-write-live-image
+[eib]: http://github.com/endlessm/eos-image-builder
+
 
-It is derived from `gnome-initial-setup`, the first boot experience for GNOME
-(and Endless OS), but the majority of the `gnome-initial-setup` code has been
-excised.
+Installing with Standalone Image
+--------------------------------
 
+We also publish an `eosinstaller` ostree & OS image. This is a cut-down build
+of the OS which excludes almost all the normal applications: it can only be
+used to run eos-installer (this application) from the `gnome-initial-setup`
+greeter session.
 
-Modes
------
+The expected partition layout in this case is: EFI and BIOS boot partitions, an
```

Comment:
Is this still accurate after #87? As far as I can tell, it's just an EFI partition + a root partition of some type.

Reply:
Actually this is still technically correct because it refers to the partition layout of the `eosinstaller` image, not the OS image it is installing, but I'll reword to make this clearer.